### PR TITLE
(BIDS-1535) Catch out of range error

### PR DESCRIPTION
**Pull request by Eisei24**
### <samp>🤖 Generated by Copilot at 79e21c3</samp>

This pull request enhances the reliability and robustness of the eth1 indexer and the `GetBlock` function. It adds a timeout mechanism for the indexer, and improves the error handling and validation of the `rpc/erigon.go` file.

### CODE REVIEW

File: rpc/erigon.go [hunk 1]
```diff
@@ -251,6 +250,10 @@ func (client *ErigonClient) GetBlock(number int64) (*types.Eth1Block, *types.Get
 				continue
 			}
 
+			if len(c.Transactions) <= trace.TransactionPosition {
```
[the remark this answers is not in the document]
Comment (Eisei24) [reply rpc/erigon.go:253]: Done.

[the remark this answers is not in the document]
Comment (recy21) [reply rpc/erigon.go:253]: no?

**Review by recy21 — CHANGES_REQUESTED** [review]

File: cmd/eth1indexer/main.go [hunk 2]
```diff
@@ -280,8 +281,16 @@ func main() {
 
 			err = IndexFromNode(bt, client, int64(lastBlockFromBlocksTable)-*offsetBlocks, int64(lastBlockFromNode), *concurrencyBlocks)
 			if err != nil {
-				logrus.WithError(err).Errorf("error indexing from node, start: %v end: %v concurrency: %v", int64(lastBlockFromBlocksTable)-*offsetBlocks, int64(lastBlockFromNode), *concurrencyBlocks)
+				errMsg := fmt.Sprintf("error indexing from node, start: %v end: %v concurrency: %v", int64(lastBlockFromBlocksTable)-*offsetBlocks, int64(lastBlockFromNode), *concurrencyBlocks)
+				errEntry := logrus.WithError(err)
+				if time.Since(lastSuccessulBlockIndexingTs) > time.Minute*30 {
+					errEntry.Fatal(errMsg)
```
Comment (recy21) [inline cmd/eth1indexer/main.go:287]: @Eisei24 we should use your utils.LogError here (as I already merged)

**Review by recy21 — CHANGES_REQUESTED** [review]

File: cmd/eth1indexer/main.go (hunk 2 above)
Comment (Eisei24) [reply cmd/eth1indexer/main.go]: Merged master into the branch and changed this part here.

File: rpc/erigon.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Eisei24) [reply rpc/erigon.go]: The commit was done a bit after the "Done" comment was written.

File: cmd/eth1indexer/main.go [hunk 3]
```diff
@@ -280,8 +281,15 @@ func main() {
 
 			err = IndexFromNode(bt, client, int64(lastBlockFromBlocksTable)-*offsetBlocks, int64(lastBlockFromNode), *concurrencyBlocks)
 			if err != nil {
-				logrus.WithError(err).Errorf("error indexing from node, start: %v end: %v concurrency: %v", int64(lastBlockFromBlocksTable)-*offsetBlocks, int64(lastBlockFromNode), *concurrencyBlocks)
+				errMsg := fmt.Sprintf("error indexing from node, start: %v end: %v concurrency: %v", int64(lastBlockFromBlocksTable)-*offsetBlocks, int64(lastBlockFromNode), *concurrencyBlocks)
```
Comment (recy21) [inline cmd/eth1indexer/main.go:284]: I would split this in 4 parameter directly as fields in the error message.

**Review by recy21 — CHANGES_REQUESTED** [review]

File: cmd/eth1indexer/main.go (hunk 3 above)
Comment (Eisei24) [reply cmd/eth1indexer/main.go]: Done.

Comment (Yokymia32) [reply cmd/eth1indexer/main.go]: Nice 👍

BIDS-1535/FatalErrorForEth1indexer

**Review by recy21 — APPROVED** [review]

**Pull request merged**